auth/auth_test.go

```diff
@@ -2648,3 +2648,33 @@ func TestInvalidateChannels(t *testing.T) {
 		})
 	}
 }
+
+func TestCalculateMaxHistoryEntriesPerGrant(t *testing.T) {
+	testCases := []struct {
+		Name                     string
+		HistoryLengthInput       int
+		MaxEntriesExpectedOutput int
+	}{
+		{
+			Name:                     "Extreme history length",
+			HistoryLengthInput:       1000000,
+			MaxEntriesExpectedOutput: 1,
+		},
+		{
+			Name:                     "Large history length",
+			HistoryLengthInput:       1000,
+			MaxEntriesExpectedOutput: 10,
+		},
```

Comment:
Include these so:
- We can see where the distribution is (these numbers are ballpark, they have a fair amount of leeway in them)
- The test fails if we change things to affect the distribution too significantly

```suggestion
		{
			Name:                     "Upper bound",
			HistoryLengthInput:       3600,
			MaxEntriesExpectedOutput: 1,
		},
		{
			Name:                     "Large history length",
			HistoryLengthInput:       3330,
			MaxEntriesExpectedOutput: 3,
		},
		{
			Name:                     "Median history length",
			HistoryLengthInput:       3075,
			MaxEntriesExpectedOutput: 5,
		},
		{
			Name:                     "Small history length",
			HistoryLengthInput:       2800,
			MaxEntriesExpectedOutput: 7,
		},
		{
			Name:                     "Lower bound",
			HistoryLengthInput:       2550,
			MaxEntriesExpectedOutput: 10,
		},
```